pkg/webhooks/sgx/sgx.go

```diff
@@ -40,9 +40,39 @@ const (
 	encl                = namespace + "/enclave"
 	epc                 = namespace + "/epc"
 	provision           = namespace + "/provision"
-	provisionAnnotation = namespace + "/needs-provision"
+	quoteProvAnnotation = namespace + "/quote-provider"
+	aesmdQuoteProvKey   = "aesmd"
 )
 
+func getAesmdVolume(needsAesmd bool, epcUserCount int32, aesmdPresent bool) *corev1.Volume {
+	switch {
+	case !needsAesmd:
+		return nil
+	case epcUserCount == 0:
+		return nil
+	case aesmdPresent && epcUserCount >= 2:
```

Comment:
I think this logic needs some comments still :-) Does this mean that aesmd will anyway be one epcUser, so this means that there is some other container using epc which requires the volume to be there?

Reply:
I'll add more commentes. emptydir volume is added when aesmd is deployed as a sidecar: the pod has aesmd container (an epcuser too) and other containers requesting sgx. In this scenario, we cannot have host mounts.

Reply:
@ipuustin added comments to explain this